pkg/target/host.go

```diff
@@ -156,14 +154,18 @@ func (h *Host) Flush(d time.Duration) {
 		case <-h.stop:
 			return
 		case <-t.C:
-			h.Wm.Lock()
-			if h.W != nil {
-				err := h.W.Flush()
-				if err != nil {
-					h.Lg.Error("error while flushing the host buffer", zap.Error(err), zap.String("host name", h.Name), zap.Uint16("host port", h.Port))
+			h.CWm.Lock()
+			if h.Conn != nil && h.W != nil {
+				if h.W.Buffered() != 0 {
```

Comment:
As you wish, this is what I would do:
```suggestion
			if h.Conn != nil && h.W != nil && h.W.Buffered() != 0 {
```

Reply:
Yes, your code in nicer! It is like that because it's a fossil from me playing around with mutexes before.